lib/spack/spack/cmd/export.py

```diff
@@ -0,0 +1,196 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import sys
+import ruamel.yaml as yaml
+
+import llnl.util.tty as tty
+import llnl.util.lang
+
+import spack.repo
+import spack.cmd.common.arguments as arguments
+from spack.cmd import display_specs
+
+from spack.build_systems.python import PythonPackage
+
+import spack.util.spack_yaml as syaml
+from spack.util.spack_yaml import syaml_dict, syaml_list
+
+description = "create a package.yaml from installed packages"
+section = "administration"
+level = "long"
+
+
+class PackagesDumper(syaml.OrderedLineDumper):
+    """Customization to match common packages.yaml style
+    """
+
+    def represent_list(self, seq, flow_style=None):
+        """Impose an arbitrary length limit up to flow lists
+        """
+        res = super(PackagesDumper, self).represent_list(seq)
+        tot_len = sum(len(e.value) for e in res.value)
+        res.flow_style = tot_len < 60
+        return res
+
+
+PackagesDumper.add_representer(syaml_list, PackagesDumper.represent_list)
+
+
+def setup_parser(subparser):
+    scopes = spack.config.scopes()
+    subparser.add_argument('-f', '--format',
+                           help="specify format for path/module keys",
+                           metavar="FMT", default='$_$@')
+    subparser.add_argument('-d', '--dependencies',
+                           help="add selected dependencies to the specs",
+                           action='store_true')
```

Comment:
when I specify dependencies but not spec itself, I get:

```
→ spack export --scope user -d mpich
packages:
  mpich:
    paths:
      mpich@3.2: /usr/local/Cellar/mpich/3.2.1_2
```
I thought above will return {} as I didnt specify spec